pkg/executer/executer_http.go

```diff
@@ -160,21 +161,50 @@ func (e *HTTPExecuter) ExecuteHTTP(ctx context.Context, p progress.IProgress, re
 		e.bulkHTTPRequest.Increment(reqURL)
 		p.Update()
 		remaining--
+
+		if len(dynvars) > 0 {
+			for extractor, dynvar := range dynvars {
+				for _, dyn := range dynvar {
+					dynamicvalues[extractor] = dyn
+					httpRequest, err := e.bulkHTTPRequest.MakeHTTPRequest(ctx, reqURL, dynamicvalues, e.bulkHTTPRequest.Current(reqURL))
+					if err != nil {
+						result.Error = errors.Wrap(err, "could not build http request")
+
+						p.Drop(remaining)
+
+						return
+					}
+
+					err, dynvars = e.handleHTTP(reqURL, httpRequest, dynamicvalues, &result)
+					if err != nil {
+						result.Error = errors.Wrap(err, "could not handle http request")
+
+						p.Drop(remaining)
+
+						return
+					}
+				}
+			}
+			e.bulkHTTPRequest.Increment(reqURL)
+			p.Update()
+			remaining--
+		}
```

Comment:
This method needs to be refactored to avoid duplication while handling HTTP requests, maybe using recursion? the warning about ineffectual assignment is the product of a missuse of `dynvars`.

Reply:
Thanks @vzamanillo, I'm still on the go-learning path.
Cheers!

/ Casper

Reply:
Thank you @CasperGN, don't worry, and don't hurry

Reply:
Thanks @vzamanillo, I'll look into a refactor and look deeper on recursion!

/ Casper

Reply:
I've tried to refactor by the latest pull. I've tried recursive function at different locations but I either end up not being able to "increment" to next request (e.g. the RAW request) properly or missing the full extraction becuase of too early increment past the RAW request.

Any pointers or ideas would be appriciated.
I've tested the current against some BB programs and i seem to get results when using templates from nuclei-templates/pull/471

/ Casper

Reply:
@CasperGN I need time to test your branch, don't think I forget about this.